src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Source/AnalyzeDiagnostics.cs

```diff
@@ -0,0 +1,133 @@
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Source
+{
+    public class AnalyzeDiagnostics : IAnalyzeResultProvider
+    {
+        private readonly IEnumerable<DiagnosticAnalyzer> _allAnalyzers;
+        private readonly ImmutableArray<AdditionalText> _additionalTexts;
+        private readonly ILogger _logger;
+
+        public string Id => "UA102";
+
+        public string Name => "API Upgradability";
+
+        public Uri InformationURI => new("https://docs.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-overview");
+
+        public AnalyzeDiagnostics(IEnumerable<DiagnosticAnalyzer> analyzers, IEnumerable<AdditionalText> additionalTexts, ILogger<AnalyzeDiagnostics> logger)
+        {
+            if (additionalTexts is null)
+            {
+                throw new ArgumentNullException(nameof(additionalTexts));
+            }
+
+            if (logger is null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            _logger = logger;
+            _allAnalyzers = analyzers;
+            _additionalTexts = ImmutableArray.CreateRange(additionalTexts);
+        }
+
+        public async IAsyncEnumerable<AnalyzeResult> AnalyzeAsync(AnalyzeContext analysis, [System.Runtime.CompilerServices.EnumeratorCancellation] CancellationToken token)
+        {
+            if (analysis is null)
+            {
+                throw new ArgumentNullException(nameof(analysis));
+            }
+
+            var context = analysis.UpgradeContext;
+            var projects = context.Projects.ToList();
+
+            foreach (var project in projects)
+            {
+                var diagnostics = await this.GetDiagnosticsAsync(project, token).ConfigureAwait(false);
+
+                yield return new()
+                {
+                    Results = ProcessDiagnostics(diagnostics),
+                };
+            }
+        }
+
+        private static HashSet<ResultObj> ProcessDiagnostics(IEnumerable<Diagnostic> diagnostics)
+        {
+            var results = new HashSet<ResultObj>();
+            foreach (var diag in diagnostics)
+            {
+                results.Add(new()
+                {
+                    // Since the first line in a file is defined as line 0 (zero based line
+                    // numbering) by the LinePostion struct offsetting by one to support VS 1-based line numbering.
+                    LineNumber = diag.Location.GetLineSpan().Span.End.Line + 1,
+                    FileLocation = diag.Location.GetLineSpan().Path,
+                    ResultMessage = diag.Descriptor.Description.ToString(System.Globalization.CultureInfo.InvariantCulture),
+                });
+            }
+
+            return results;
+        }
+
+        private static IAsyncEnumerable<DiagnosticAnalyzer> GetApplicableAnalyzersAsync(IEnumerable<DiagnosticAnalyzer> analyzers, IProject project)
```

Comment:
This doesn't do anything - none of the analyzers have the components anymore. If you'd like to see that, we can add an issue, but this logic here should be centralized rather than a private method here.

Reply:
Removed the filter